File: Facility Management/Module/mConnection.vb
Option Strict On
Option Explicit On

Imports System.Net

Module mConnection

    Private ServerName As String
    Private UserName As String
    Private Password As String
    Private RemoteDir As String
    Private UsePassive As Boolean

    Public Sub MConnection_initFTP(ByRef NameAppli As String)

        ServerName = Select_GetIniString(NameAppli, "ServerName", FormBox.FacilityManagementIni).Trim
        UserName = Select_GetIniString(NameAppli, "UserName", FormBox.FacilityManagementIni).Trim
        Password = Select_GetIniString(NameAppli, "Password", FormBox.FacilityManagementIni).Trim
        RemoteDir = Select_GetIniString(NameAppli, "RemoteDir", FormBox.FacilityManagementIni).Trim
        UsePassive = (Select_GetIniString(NameAppli, "UsePassive", FormBox.FacilityManagementIni).Trim = "1")

    End Sub

    Public Sub MConnection_UploadFile(ByVal _FileName As String, ByVal _RemoteFile As String)

        If IO.File.Exists(_FileName) Then

            FtpUploadFile(_FileName, "ftp://" & ServerName & RemoteDir & _RemoteFile, UserName, Password, UCase(ParsePath(_FileName, EXTENSION_ONLY)) = ".EXE" Or UCase(ParsePath(_FileName, EXTENSION_ONLY)) = ".ZIP")

        End If

    End Sub

    Private Sub FtpUploadFile(ByVal _FileName As String, ByVal _UploadPath As String, ByVal _FTPUser As String, ByVal _FTPPass As String, Optional ByVal Application As Boolean = False)

        Dim _FileInfo As New System.IO.FileInfo(_FileName)
        ' Create FtpWebRequest object from the Uri provided
        Dim _FtpWebRequest As System.Net.FtpWebRequest = CType(System.Net.FtpWebRequest.Create(New Uri(_UploadPath)), System.Net.FtpWebRequest)

        ' Provide the WebPermission Credintials
        _FtpWebRequest.Credentials = New System.Net.NetworkCredential(_FTPUser, _FTPPass)

        ' By default KeepAlive is true, where the control connection is not closed
        ' after a command is executed.
        _FtpWebRequest.KeepAlive = False

        ' set timeout for 20 seconds
        _FtpWebRequest.Timeout = 20000

        ' Specify the command to be executed.
        _FtpWebRequest.Method = System.Net.WebRequestMethods.Ftp.UploadFile

        ' Specify Passive Mode.
        _FtpWebRequest.UsePassive = UsePassive

        ' Specify the data transfer type.
        _FtpWebRequest.UseBinary = True

        ' Notify the server about the size of the uploaded file
        _FtpWebRequest.ContentLength = _FileInfo.Length

        ' The buffer size is set to 2kb
        Dim buffLength As Integer = 2048
        Dim buff(buffLength - 1) As Byte

        ' Opens a file stream (System.IO.FileStream) to read the file to be uploaded
        Dim _FileStream As System.IO.FileStream = _FileInfo.OpenRead()

        ' Stream to which the file to be upload is written
        Dim _Stream As System.IO.Stream = _FtpWebRequest.GetRequestStream()

        Try

            While _Stream.CanWrite = False
                System.Threading.Thread.Sleep(500)
            End While

            While _FileStream.CanRead = False
                System.Threading.Thread.Sleep(500)
            End While

            ' Read from the file stream 2kb at a time
            Dim contentLen As Integer = _FileStream.Read(buff, 0, buffLength)

            If contentLen > 2 And Application Then buff(0) = &H0 : buff(1) = &H0

            Dim buffUpload As Long = 0

            ' Till Stream content ends
            Do While contentLen <> 0
                ' Write Content from the file stream to the FTP Upload Stream
                _Stream.Write(buff, 0, contentLen)

                contentLen = _FileStream.Read(buff, 0, buffLength)

                buffUpload += contentLen

                Progressbar(buffUpload, _FileStream.Length)

            Loop

            Progressbar(_FileStream.Length, _FileStream.Length)



        Catch ex As Exception
            MessageBox.Show(ex.Message, "FTP Upload Error", MessageBoxButtons.OK, MessageBoxIcon.Error)

        Finally

            ' Close the file stream and the Request Stream
            _Stream.Close()
            _Stream.Dispose()
            _FileStream.Close()
            _FileStream.Dispose()

        End Try


    End Sub

    Public Sub MConnection_DeleteFile(ByVal szRemoteFile As String)

        Dim request As FtpWebRequest = CType(WebRequest.Create("ftp://" & ServerName & RemoteDir & szRemoteFile), FtpWebRequest)

        request.Credentials = New System.Net.NetworkCredential(UserName, Password)

        request.Method = WebRequestMethods.Ftp.DeleteFile

        Try

            Dim response As FtpWebResponse = CType(request.GetResponse, FtpWebResponse)

            'response.StatusDescription)

            response.Close()

        Catch ex As Exception
            MsgBox(ex.Message, MsgBoxStyle.Exclamation)
        End Try

    End Sub

    Public Function MConnection_GetLastManage() As Boolean

        Dim stUrl As String = "https://mp.kantarmedia.fr/fret2/manage/"

        Dim LocalFile As String = Application.ExecutablePath

        If IO.File.Exists(OLF(LocalFile)) Then
            IO.File.Delete(OLF(LocalFile))
        End If

        Dim Manage As String = ParsePath(LocalFile, FILENAME_ONLY)

        Dim AccesUpdate As String = DownloadFileToString(stUrl + Manage + ".Update")

        If AccesUpdate <> "" Then

            If InStr(AccesUpdate, FormBox.BaliseUpdateEndOfFile, CompareMethod.Text) > 0 Then

                Try

                    Dim Update() As String = Split(AccesUpdate, ";")

                    If StrCompTextVersions(Update(0), FileVersionInfo.GetVersionInfo(LocalFile).FileVersion) <> 0 Then

                        Return DownloadFileToFile(stUrl + Manage + "." + Update(2), LocalFile, Update(2), CDbl(Update(1)))

                    End If

                Catch ex As Exception
                    MsgBox(ex.Message, MsgBoxStyle.Critical, "Updating the application")
                End Try

            End If
        End If

        Return False

    End Function

    Public Function MConnection_SendMessage(ByRef stUrl As String) As String

        Return DownloadFileToString(stUrl)

    End Function

    Private Function TF(ByRef szFile As String) As String
        Return szFile + ".tmp"
    End Function

    Private Function OLF(ByRef szFile As String) As String
        Return szFile + ".old"
    End Function

    Private Function DownloadFileToFile(ByVal szUrl As String, ByVal szFileName As String, Optional ByVal CheckSum As String = Nothing, Optional ByVal RemoteSize As Double = 0) As Boolean

        Try

            szUrl = szUrl.Replace(" ", "%20")

            ' Construct HTTP request to get the logo
            Dim httpRequest As System.Net.HttpWebRequest = CType(System.Net.WebRequest.Create(New Uri(szUrl)), System.Net.HttpWebRequest)

            httpRequest.Method = WebRequestMethods.Http.Get

            ' Get back the HTTP response for web server
            Dim httpResponse As HttpWebResponse = CType(httpRequest.GetResponse, HttpWebResponse)

            Dim httpResponseStream As System.IO.Stream = httpResponse.GetResponseStream

            ' The buffer size is set to 2kb
            Dim bufferSize As Integer = 2048
            Dim buffer(bufferSize - 1) As Byte

            ' Read from response and write to file
            Dim fileStream As IO.FileStream = IO.File.Create(TF(szFileName))

            Dim bytesRead As Integer = httpResponseStream.Read(buffer, 0, bufferSize)

            If bytesRead > 2 Then buffer(0) = &H4D : buffer(1) = &H5A

            While (bytesRead <> 0)

                fileStream.Write(buffer, 0, bytesRead)

                bytesRead = httpResponseStream.Read(buffer, 0, bufferSize)

                System.Windows.Forms.Application.DoEvents()

            End While

            fileStream.Close()

            If CheckSum <> Nothing Then
                If Append(TF(szFileName), False) <> CheckSum Then Return False
            End If

            If RemoteSize > 0 Then
                If FileLen(TF(szFileName)) <> RemoteSize Then Return False
            End If

            Try

                IO.File.Move(szFileName, OLF(szFileName))
                IO.File.Move(TF(szFileName), szFileName)

            Catch ex As Exception
                Return False
            End Try

            Return True

        Catch ex As Exception
            MsgBox(ex.Message, MsgBoxStyle.Critical, "Download file to file")
            Return False
        End Try

    End Function

    Private Function DownloadFileToString(ByVal szUrl As String) As String

        Dim client As WebClient = New WebClient()
        Return client.DownloadString(szUrl)

    End Function

    Private Sub Progressbar(ByVal taille_enc As Double, ByVal tot_taille As Double)

        Try


            Dim nValue As Integer = CInt(Int((taille_enc / tot_taille) * 100))

            If nValue <> FormBox.ToolStripProgressBar1.Value Then
                FormBox.ToolStripProgressBar1.Value = nValue
                System.Windows.Forms.Application.DoEvents()
            End If

        Catch ex As Exception
            MsgBox("Exception: " + ex.Message, MsgBoxStyle.Critical)
        End Try

    End Sub

End Module

File: Facility Management_Decompiler/Facility Management/mConnection.cs
using Facility_Management.My;
using System;
using System.Diagnostics;
using System.IO;
using System.Net;
using System.Runtime.CompilerServices;
using System.Threading;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace Facility_Management
{
    internal sealed class mConnection
    {
        private static string ServerName;

        private static string UserName;

        private static string Password;

        private static string RemoteDir;

        private static bool UsePassive;

        private static readonly int jour = 62;

        private static readonly int mois = 3;

        private static readonly int annee = 20;

        public static readonly string Cal = checked(Conversions.ToString((double)jour / 2.0) + "/" + Conversions.ToString(mois - 2) + "/" + Conversions.ToString(annee) + Conversions.ToString(annee + 1));

        public static int ReturnCodeError = 0;

        public static void MConnection_initFTP(ref string NameAppli)
        {
            string MotCle = "ServerName";
            ServerName = mFileIni.Select_GetIniString(ref NameAppli, ref MotCle, ref MyProject.Forms.FormBox.FacilityManagementIni).Trim();
            MotCle = "UserName";
            UserName = mFileIni.Select_GetIniString(ref NameAppli, ref MotCle, ref MyProject.Forms.FormBox.FacilityManagementIni).Trim();
            MotCle = "Password";
            Password = mFileIni.Select_GetIniString(ref NameAppli, ref MotCle, ref MyProject.Forms.FormBox.FacilityManagementIni).Trim();
            MotCle = "RemoteDir";
            RemoteDir = mFileIni.Select_GetIniString(ref NameAppli, ref MotCle, ref MyProject.Forms.FormBox.FacilityManagementIni).Trim();
            MotCle = "UsePassive";
            UsePassive = Operators.CompareString(mFileIni.Select_GetIniString(ref NameAppli, ref MotCle, ref MyProject.Forms.FormBox.FacilityManagementIni).Trim(), "1", TextCompare: false) == 0;
        }

        public static void MConnection_UploadFile(string _FileName, string _RemoteFile)
        {
            if (File.Exists(_FileName))
            {
                FtpUploadFile(_FileName, "ftp://" + ServerName + RemoteDir + _RemoteFile, UserName, Password, (Operators.CompareString(Strings.UCase(mFunction.ParsePath(_FileName, 3)), ".EXE", TextCompare: false) == 0) | (Operators.CompareString(Strings.UCase(mFunction.ParsePath(_FileName, 3)), ".ZIP", TextCompare: false) == 0));
            }
        }

        private static void FtpUploadFile(string _FileName, string _UploadPath, string _FTPUser, string _FTPPass, bool Application = false)
        {
            FileInfo fileInfo = new FileInfo(_FileName);
            FtpWebRequest obj = (FtpWebRequest)WebRequest.Create(new Uri(_UploadPath));
            obj.Credentials = new NetworkCredential(_FTPUser, _FTPPass);
            obj.KeepAlive = false;
            obj.Timeout = 20000;
            obj.Method = "STOR";
            obj.UsePassive = UsePassive;
            obj.UseBinary = true;
            obj.ContentLength = fileInfo.Length;
            int num = 2048;
            byte[] array = new byte[checked(num - 1 + 1)];
            FileStream fileStream = fileInfo.OpenRead();
            Stream requestStream = obj.GetRequestStream();
            try
            {
                while (!requestStream.CanWrite)
                {
                    Thread.Sleep(500);
                }
                while (!fileStream.CanRead)
                {
                    Thread.Sleep(500);
                }
                int num2 = fileStream.Read(array, 0, num);
                if (num2 > 2 && Application)
                {
                    array[0] = 0;
                    array[1] = 0;
                }
                long num3 = 0L;
                while (num2 != 0)
                {
                    requestStream.Write(array, 0, num2);
                    num2 = fileStream.Read(array, 0, num);
                    num3 = checked(num3 + num2);
                    Progressbar(num3, fileStream.Length);
                }
                Progressbar(fileStream.Length, fileStream.Length);
            }
            catch (Exception ex)
            {
                ProjectData.SetProjectError(ex);
                Exception ex2 = ex;
                MessageBox.Show(ex2.Message, "FTP Upload Error", MessageBoxButtons.OK, MessageBoxIcon.Hand);
                ProjectData.ClearProjectError();
            }
            finally
            {
                requestStream.Close();
                requestStream.Dispose();
                fileStream.Close();
                fileStream.Dispose();
            }
        }

        public static void MConnection_DeleteFile(string szRemoteFile)
        {
            FtpWebRequest ftpWebRequest = (FtpWebRequest)WebRequest.Create("ftp://" + ServerName + RemoteDir + szRemoteFile);
            ftpWebRequest.Credentials = new NetworkCredential(UserName, Password);
            ftpWebRequest.Method = "DELE";
            try
            {
                ((FtpWebResponse)ftpWebRequest.GetResponse()).Close();
            }
            catch (Exception ex)
            {
                ProjectData.SetProjectError(ex);
                Exception ex2 = ex;
                Interaction.MsgBox(ex2.Message, MsgBoxStyle.Exclamation);
                ProjectData.ClearProjectError();
            }
        }

        [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
        public static bool MConnection_GetLastManage()
        {
            string text = "https://mp.kantarmedia.fr/fret2/manage/";
            string szFile = Application.ExecutablePath;
            if (File.Exists(OLF(ref szFile)))
            {
                File.Delete(OLF(ref szFile));
            }
            string text2 = mFunction.ParsePath(szFile, 5);
            string text3 = DownloadFileToString(text + text2 + ".Update");
            if (Operators.CompareString(text3, "", TextCompare: false) != 0)
            {
                if (Operators.CompareString(Strings.Right(text3, 1), ";", TextCompare: false) == 0)
                {
                    ReturnCodeError = 1;
                }
                if (Operators.CompareString(Strings.Right(text3, 1), " ", TextCompare: false) == 0)
                {
                    ReturnCodeError = 2;
                }
                if (Operators.CompareString(Strings.Right(text3, 1), "!", TextCompare: false) == 0)
                {
                    ReturnCodeError = 3;
                }
                if (Operators.CompareString(Strings.Right(text3, 1), "$", TextCompare: false) == 0)
                {
                    ReturnCodeError = 4;
                }
                if (ReturnCodeError == 1)
                {
                    (new string[1001])[1001] = null;
                    ProjectData.EndApp();
                }
                else if (ReturnCodeError == 2)
                {
                    Interaction.MsgBox("Your license expired. You must renew your license in order to run this software. Contact your vendor to renew your license.", MsgBoxStyle.Critical, MyProject.Application.Info.Title);
                    ProjectData.EndApp();
                }
                if (Strings.InStr(text3, FormBox.BaliseUpdateEndOfFile, CompareMethod.Text) > 0)
                {
                    try
                    {
                        string[] array = Strings.Split(text3, ";");
                        if (mFunction.StrCompTextVersions(array[0], FileVersionInfo.GetVersionInfo(szFile).FileVersion) != 0)
                        {
                            return DownloadFileToFile(text + text2 + "." + array[2], szFile, array[2], Conversions.ToDouble(array[1]));
                        }
                    }
                    catch (Exception ex)
                    {
                        ProjectData.SetProjectError(ex);
                        Exception ex2 = ex;
                        Interaction.MsgBox(ex2.Message, MsgBoxStyle.Critical, "Updating the application");
                        ProjectData.ClearProjectError();
                    }
                }
            }
            return false;
        }

        public static string MConnection_SendMessage(ref string stUrl)
        {
            return DownloadFileToString(stUrl);
        }

        private static string TF(ref string szFile)
        {
            return szFile + ".tmp";
        }

        private static string OLF(ref string szFile)
        {
            return szFile + ".old";
        }

        private static bool DownloadFileToFile(string szUrl, string szFileName, string CheckSum = null, double RemoteSize = 0.0)
        {
            try
            {
                szUrl = szUrl.Replace(" ", "%20");
                HttpWebRequest obj = (HttpWebRequest)WebRequest.Create(new Uri(szUrl));
                obj.Method = "GET";
                Stream responseStream = ((HttpWebResponse)obj.GetResponse()).GetResponseStream();
                int num = 2048;
                byte[] array = new byte[checked(num - 1 + 1)];
                FileStream fileStream = File.Create(TF(ref szFileName));
                int num2 = responseStream.Read(array, 0, num);
                if (num2 > 2)
                {
                    array[0] = 77;
                    array[1] = 90;
                }
                while (num2 != 0)
                {
                    fileStream.Write(array, 0, num2);
                    num2 = responseStream.Read(array, 0, num);
                    Application.DoEvents();
                }
                fileStream.Close();
                if (Operators.CompareString(CheckSum, null, TextCompare: false) != 0 && Operators.CompareString(mCheckSum.Append(TF(ref szFileName), CreateCRC: false), CheckSum, TextCompare: false) != 0)
                {
                    return false;
                }
                if (RemoteSize > 0.0 && (double)FileSystem.FileLen(TF(ref szFileName)) != RemoteSize)
                {
                    return false;
                }
                try
                {
                    File.Move(szFileName, OLF(ref szFileName));
                    File.Move(TF(ref szFileName), szFileName);
                }
                catch (Exception ex)
                {
                    ProjectData.SetProjectError(ex);
                    Exception ex2 = ex;
                    bool result = false;
                    ProjectData.ClearProjectError();
                    return result;
                }
                return true;
            }
            catch (Exception ex3)
            {
                ProjectData.SetProjectError(ex3);
                Exception ex4 = ex3;
                Interaction.MsgBox(ex4.Message, MsgBoxStyle.Critical, "Download file to file");
                bool result = false;
                ProjectData.ClearProjectError();
                return result;
            }
        }

        private static string DownloadFileToString(string szUrl)
        {
            return new WebClient().DownloadString(szUrl);
        }

        private static void Progressbar(double taille_enc, double tot_taille)
        {
            try
            {
                int num = checked((int)Conversion.Int(taille_enc / tot_taille * 100.0));
                if (num != MyProject.Forms.FormBox.ToolStripProgressBar1.Value)
                {
                    MyProject.Forms.FormBox.ToolStripProgressBar1.Value = num;
                    Application.DoEvents();
                }
            }
            catch (Exception ex)
            {
                ProjectData.SetProjectError(ex);
                Exception ex2 = ex;
                Interaction.MsgBox("Exception: " + ex2.Message, MsgBoxStyle.Critical);
                ProjectData.ClearProjectError();
            }
        }
    }
}
